internal/server/singleprocess/auth.go

```diff
@@ -293,27 +349,27 @@ func (s *service) newToken(
 func (s *service) GenerateInviteToken(
 	ctx context.Context, req *pb.InviteTokenRequest,
 ) (*pb.NewTokenResponse, error) {
+	currentUser := s.userFromContext(ctx)
+
 	// Old behavior, if we have the entrypoint set, we convert that to
 	// a request in the new (WP 0.5+) style. We do this right away so the rest of the
 	// request can assume the new style.
 	if ep := req.UnusedEntrypoint; ep != nil {
 		req.Signup = nil // not a signup token
 		req.Login = &pb.Token_Login{
-			UserId:     DefaultUser,
+			UserId:     DefaultUserId,
 			Entrypoint: ep,
 		}
 	}
 
 	if req.Login == nil {
 		req.Login = &pb.Token_Login{
-			// TODO(mitchellh): when we have a login system, set this
-			// to the currently logged in user.
-			UserId: DefaultUser,
+			UserId: currentUser.Id,
 		}
 	}
 
 	// We don't currently have any other users.
```

Comment:
Can probably remove this comment

Reply:
You actually found a missing feature. I have to implement this.

Reply:
Fixed: https://github.com/hashicorp/waypoint/pull/1752/commits/0aebac367ca51b8f7c4c377167011487df517325